nornir/core/inventory.py

```diff
@@ -428,6 +428,17 @@ def filter(self, filter_obj=None, filter_func=None, *args, **kwargs):
     def __len__(self):
         return self.hosts.__len__()
 
+    def _update_group_refs(self, inventory_element: InventoryElement) -> None:
+        """
+        Returns inventory_element with updated group references for the supplied
+        inventory element
+        """
+        if hasattr(inventory_element, "groups"):
```

Comment:
I added the `if hasattr` because it was throwing an `AttributeError` if there were no pre-defined groups in the inventory. This addresses that without the need for implementing error handling.